pkg/azuredisk/nodeserver.go

```diff
@@ -326,20 +326,28 @@ func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (
 
 	if d.supportZone {
 		var (
-			zone      cloudprovider.Zone
-			zoneError error
+			zone cloudprovider.Zone
+			err  error
 		)
-		if runtime.GOOS == "windows" && (!d.cloud.UseInstanceMetadata || d.cloud.Metadata == nil) {
-			zone, zoneError = d.cloud.VMSet.GetZoneByNodeName(d.NodeID)
+
+		if d.getNodeInfoFromLabels {
+			zone.FailureDomain, _, err = getNodeInfoFromLabels(ctx, d.NodeID, d.cloud.KubeClient)
```

Comment:
This is a nit that was noticed by @msau42 while reviewing the internal PR, we're making 2 calls to the API server, in the proposed WIP PR I think I made a call to it once and with the returned node object I the properties needed e.g. the failure domain label here and the instance type later.

An additional call to the API server is not a concern though.

Reply:
with https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1200/commits/8bfdfe4db6cb0937391070fadaf256da10cbd489, there is only one call to api server now, PTAL.

Reply:
thanks for making the changes, I added this comment https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1200#discussion_r819297469 and I think that the logic is good.

/remove-hold